### ISSUE TITLE
Rate limit causing HTTP Error 429

### ISSUE THREAD

**Issue by Marekkon5** [ramsayleung/rspotify#309]
**Describe the bug**
You can still hit the rate limit, which causes HTTP status 429.

**To Reproduce**
Make a loop doing single request such as search. This is kinda hard to reproduce, because it depends on your internet connection, and also luck (Spotify servers might be busy = slower = won't reach the rate limit). In our case it happened randomly after spamming the API for few hours. 

**Expected behavior**
`rspotify` shouldn't hit it, or retry in case it happens, not return an error.

**Log/Output data**
`http error: status code 429`

**Additional context**
I don't know where the rate limit handling is implemented or how it's implemented.

**Comment by marioortizmanero** [comment]
I honestly don't think rate limiting is something rspotify should handle. If your client gets one of these, just wait for the indicated amount of time and retry again.

**Comment by Marekkon5** [comment]
In that case why isn't there any helper method or error type which would contain the `Retry-After` amount to make it more user friendly?

**Comment by marioortizmanero** [comment]
The thing is that it was quite hard to wrap all the errors under the same interface. Since we now support both ureq and reqwest, these errors are reported in different ways depending on what client you're using. It's certainly possible, but for the first version I chose the easiest option and just left the original error types. For now you can check reqwest/ureq's documentation and see how they handle these errors yourself. We could try to wrap the errors in the future, but I remember that it was somewhat complicated when I tried.

**Comment by ramsayleung** [comment]
> I honestly don't think rate limiting is something rspotify itself should handle. If your client gets one of these, just wait for the indicated amount of time and retry again.

I agree with Mario's point, rspotify should be a lightweight rust wrapper for the Spotify Web API, instead of an all-in-on clumsy library. 

The rate limit error isn't raised from `rspotify`, and `rspotify` should not cover this up for client.
